chutney/ui/src/app/modules/campaign/components/campaign-list/campaign-list.component.ts

```diff
@@ -97,6 +97,16 @@ export class CampaignListComponent implements OnInit, OnDestroy {
         this.loadSchedulingCampaign();
     }
 
+    scheduledCampaignHasMissingEnv(scheduledCampaign: CampaignScheduling): boolean {
+        return scheduledCampaign.campaignsId.filter(campaignId => {
```

Comment:
It's more elegant like this : 
```suggestion
scheduledCampaignHasMissingEnv(scheduledCampaign: CampaignScheduling): boolean {
    return scheduledCampaign.campaignsId.some(campaignId => {
        const campaign = this.campaigns.find(campaign => campaign.id === campaignId);
        return campaign == null || campaign.environment == null;
    });
}
```
I think that `campaign == null` check `null` and `undefined`.
`some' <=> `filter` + `length > 0`